bpt_inspection.cpp

```diff
@@ -0,0 +1,114 @@
+#include <iostream>
+#include <set>
+
+#include <boost/foreach.hpp>
+
+#include "bpt_args_list.hpp"
+#include "bpt_inspection.hpp"
+
+namespace bpt {
+
+static void inspect_inst_reads(INS ins, std::set<REG>& reads) {
+    for (UINT32 i = 0, I = INS_OperandCount(ins); i < I; ++i) {
+        if (INS_OperandIsReg(ins, i)) {
+            REG reg = INS_OperandReg(ins, i);
+            REG parrent = REG_FullRegName(reg);
```

Comment:
Yes, my typo. I will replace `parrent` with `base`. Where base is whole `register`, which contains part of `reg`